javatests/arcs/core/host/test.arcs

```diff
@@ -5,9 +5,15 @@ particle TestProdParticle in 'arcs.core.host.TestProdParticle'
 
 particle TestHostParticle in 'arcs.core.host.TestHostParticle'
 
+particle TestReflectiveParticle in 'arcs.core.host.TestReflectiveParticle'
+  data: reads [Thing {name: Text}]
+
 recipe TestRecipe
```

Comment:
I think we should split this recipe into 2 separate ones:
```
recipe TestRecipe
  testHandle: create
  TestProdParticle
  TestHostParticle
```

```
recipe ReflectiveParticleRecipe
  testHandle: create
  TestReflectiveParticle
    data: reads testHandle
```

This way you don't need to register TestProdParticle and TestHostParticle in your test.